helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml

```diff
@@ -36,6 +47,103 @@ spec:
+        cluster_id: gremlin
+        cluster_type: management_cluster
+      record: raw_slo_errors
+    - expr: "vector(0.01 * 36)"
+      labels:
+        service: jose
+      record: slo_target_high
+    - expr: "vector(0.01 * 12)"
+      labels:
+        service: jose
+      record: slo_target_low
+
+    # -- managed-apps
+    # record when pods of a daemonset with label "label_giantswarm_io_monitoring_basic_sli" are down
+    - expr: |
+        kube_daemonset_status_number_unavailable
+        and on(daemonset,cluster_id,cluster_type,namespace, workload_name)
+        kube_daemonset_labels{label_giantswarm_io_monitoring_basic_sli='true'}
+      labels:
+        class: MEDIUM
+        area: managed-apps
+        service: {{ $labels.workload_name }}
+      record: raw_slo_errors
+    # record when pods of a deployment with label "label_giantswarm_io_monitoring_basic_sli" are down
+    - expr: |
+        kube_deployment_status_replicas_unavailable
+        and on(deployment,cluster_id,cluster_type,namespace, workload_name) 
+        kube_deployment_labels{label_giantswarm_io_monitoring_basic_sli='true'}
+      labels:
+        class: MEDIUM
+        area: managed-apps
+        service: {{ $labels.workload_name }}
+      record: raw_slo_errors
+    # record when pods of a statefulset with label "label_giantswarm_io_monitoring_basic_sli" are down
+    - expr: |
+        kube_statefulset_status_replicas - kube_statefulset_status_replicas_current
+        and on(statefulset,cluster_id,cluster_type,namespace, workload_name) 
+        kube_statefulset_labels{label_giantswarm_io_monitoring_basic_sli='true'}
+      labels:
+        class: MEDIUM
+        area: managed-apps
+        service: {{ $labels.workload_name }}
+      record: raw_slo_errors
+    - expr: |
+        kube_daemonset_status_desired_number_scheduled
+        and on(daemonset,cluster_id,cluster_type,namespace, workload_name)
+        kube_daemonset_labels{label_giantswarm_io_monitoring_basic_sli='true'}
+      labels:
+        class: MEDIUM
+        area: managed-apps
+        service: {{ $labels.workload_name }}
+      record: raw_slo_requests
+    - expr: |
+        kube_deployment_status_replicas
+        and on(deployment,cluster_id,cluster_type,namespace, workload_name) 
+        kube_deployment_labels{label_giantswarm_io_monitoring_basic_sli='true'}
+      labels:
+        class: MEDIUM
+        area: managed-apps
+        service: {{ $labels.workload_name }}
+      record: raw_slo_requests
+    - expr: |
+        kube_statefulset_status_replicas
+        and on(statefulset,cluster_id,cluster_type,namespace, workload_name) 
+        kube_statefulset_labels{label_giantswarm_io_monitoring_basic_sli='true'}
+      labels:
+        class: MEDIUM
+        area: managed-apps
+        service: {{ $labels.workload_name }}
+      record: raw_slo_requests
+    - expr: sum by (service) (raw_slo_errors{area="managed-apps"} - raw_slo_errors{area="managed-apps"}) + 0.05 * 36
```

Comment:
this is a hack in order to get each value of "service" for managed apps with the same slo value. Let me know if you have a more elegant way

Reply:
Can you explain the hack?

Reply:
Yeah.
So we want to have "0.05 * 36" as slo_target for all managed app services. (That is currently used in the managed service slo alerts) 
All the service names can be found in the `raw_slo_errors` metrics above (We use the "area=managed-apps' label to only get the applicable ones). 
We then substract this metric from itself so that it is always 0. Then we sum by `service` so that there is only one row for each service. The sum is also 0. Because of that we can just add "0.05 * 36" and receive the value we need.

Reply:
Alright it make sense.
Now I can see a world where we define SLA using kubernetes annotations :D